### PR TITLE
 DCA-415: Change get request method on post

### PR DESCRIPTION
**Pull request by mmizin**
https://bulldog.internal.atlassian.com/browse/DCA-415

### CODE REVIEW

File: app/util/data_preparation/api/jira_clients.py [hunk 1]
```diff
@@ -97,9 +98,15 @@ def issues_search(self, jql='order by key', start_at=0, max_results=1000, valida
         last_loop_remainder = max_results % BATCH_SIZE_ISSUES
 
         while loop_count > 0:
-            api_url = f'{self.host}/rest/api/2/search?jql={jql}&startAt={start_at}&maxResults={max_results}' \
-                      f'&validateQuery={validate_query}&fields={fields}'
-            response = self.get(api_url, "Could not retrieve issues")
+            api_url = f'{self.host}/rest/api/2/search'
```
Comment (ometelytsia) [inline app/util/data_preparation/api/jira_clients.py:101]: api_url variable is not changing in loop cycles, so it does not make sense to re-assign it on every loop. Let's move out it from a `while` statement.

File: app/util/data_preparation/api/jira_clients.py [hunk 2]
```diff
@@ -97,9 +98,15 @@ def issues_search(self, jql='order by key', start_at=0, max_results=1000, valida
...
+            pay_load = {
+                    "jql": f"{jql}",
```
Comment (ometelytsia) [inline app/util/data_preparation/api/jira_clients.py:103]: No need to use f-annotation here. Just `"jql": jql,`

File: app/util/data_preparation/api/jira_clients.py [hunk 3]
```diff
@@ -97,9 +98,15 @@ def issues_search(self, jql='order by key', start_at=0, max_results=1000, valida
         last_loop_remainder = max_results % BATCH_SIZE_ISSUES
 
         while loop_count > 0:
-            api_url = f'{self.host}/rest/api/2/search?jql={jql}&startAt={start_at}&maxResults={max_results}' \
-                      f'&validateQuery={validate_query}&fields={fields}'
-            response = self.get(api_url, "Could not retrieve issues")
+            api_url = f'{self.host}/rest/api/2/search'
+            pay_load = {
```
[the remark this answers is not in the document]
Comment (mmizin) [reply app/util/data_preparation/api/jira_clients.py:102]: I will rename itto body

File: app/util/data_preparation/api/jira_clients.py [hunk 4]
```diff
@@ -92,14 +91,23 @@ def issues_search(self, jql='order by key', start_at=0, max_results=1000, valida
         :return: Returns the requested issues
         """
 
+        if fields is None:
```
Comment (astashys) [inline app/util/data_preparation/api/jira_clients.py:94]: you can set it as default value in the function signature

File: app/util/data_preparation/api/jira_clients.py [hunk 5]
```diff
@@ -92,14 +91,23 @@ def issues_search(self, jql='order by key', start_at=0, max_results=1000, valida
...
         while loop_count > 0:
-            api_url = f'{self.host}/rest/api/2/search?jql={jql}&startAt={start_at}&maxResults={max_results}' \
-                      f'&validateQuery={validate_query}&fields={fields}'
-            response = self.get(api_url, "Could not retrieve issues")
+
+            body = {
+                    "jql": jql,
+                    "startAt": start_at,
+                    "maxResults": max_results,
+                    "fields": fields
```
Comment (astashys) [inline app/util/data_preparation/api/jira_clients.py:107]: "fields": ['id'] if fields is None else fields

**Pull request merged**